### DISCUSSION TITLE
Exposición «Planificación justa: Lotería y Linux Completely Fair Scheduler»: preguntas y respuestas

### DISCUSSION THREAD

**Discussion by gwolf** [unamfi/sistop-2022-1#169] [Q&A]
Abro este foro como un espacio de discusión para la exposición que prepararon los alumnos Armando Ugalde Velasco y Diego Santiago Gutiérrez, «Planificación justa: Lotería y Linux Completely Fair Scheduler». El video está disponible en:

https://youtu.be/xA5bRTH5VLI

Pueden consultar también el documento de texto en que los alumnos presentan este mismo tema:

https://github.com/unamfi/sistop-2022-1/blob/main/exposiciones/SantiagoDiego-UgaldeArmando/planificacion-justa.pdf

Además de participar aquí, les pido que llenen la pequña encuesta que está disponible en:

https://www.questionpro.com/t/ARiwuZqERf

**Comment by JesusESD** [comment]
¡Saludos compañeros! Su video me pareció muy completo, incluso cuando por el tema del tiempo no pudieron detenerse mucho en todos los ejemplos. Mi pregunta es con respecto a la inflación en la planificación de lotería: ¿no existe algún método que permita la existencia de la inflación un ambiente competitivo controlando los procesos voraces? ¿O directamente la inflación no es útil en esos contextos? Les agradeceré mucho su respuesta.

**Reply by armandouv** [reply]
Hola, ¡gracias por preguntar y por tus comentarios! La inflación es un mecanismo que, por su naturaleza, permite que un proceso incremente sus boletos de forma arbitraria, y, por ende, pueda monopolizar el CPU al tener muchos, como se discutió en el vídeo. Podríamos pensar en crear alguna regla para evitar que los procesos "abusen" de esto: por ejemplo, podríamos poner cierto límite en la cantidad de boletos que se pueden aumentar. Sin embargo, lo anterior no sería tan viable, ya que podríamos requerir que cierto proceso sí utilice el CPU por una proporción grande de tiempo, por lo que lo estaríamos limitando.

Si bien mencionamos un ejemplo de una regla que podría establecerse para resolver el problema que mencionas, la inflación tiene mucho más sentido en un escenario donde los procesos confían entre sí: es decir, si alguno necesita más recursos, aumenta su cantidad de boletos de forma "razonable", según el caso específico. Para el otro caso (procesos que no confían entre sí), existen algoritmos más adecuados, como la cola de retroalimentación multinivel o incluso el CFS, donde un proceso no puede asignarse a sí mismo más recursos, sino que son asignados por el planificador mismo. ¡Espero haber respondido a tu pregunta!

**Comment by Inakyord** [comment]
Hola, espero estén bien. Tanto el video como el trabajo escrito me parecieron bastante buenos. Mi pregunta es la siguiente. Mencionaron algunos valores usuales para los parámetros del CFS, ¿Estos de dónde los sacaron, son los que utiliza Linux (cómo se llegaron a estos)? Y A pesar de ser un algoritmo propio del sistema operativo, ¿El usuario/administrador puede manipular alguno de los parámetros manualmente para ajustar a su conveniencia el funcionamiento del scheduler? Gracias.

**Reply by armandouv** [reply]
Hola, ¡gracias por preguntar y por tus comentarios! Con respecto a tu primera pregunta, los valores que mencionamos como usuales para los parámetros se consultaron en un capítulo del libro Operating Systems: Three Easy Pieces, que puedes encontrar en la sección de fuentes consultadas de las diapositivas y del documento escrito. Típicamente, la forma de determinar este típo de constantes es mediante numerosas pruebas de rendimiento, es decir, no hay un método en concreto que siempre funcione para encontrar valores adecuados. De hecho, una forma de referirse a este tipo de parámetros es constantes "vudú", ya que parece que se obtienen "por arte de magia".

Por otro lado, y, respondiendo a tu segunda pregunta, los valores son "usuales" según el libro; sin embargo, claro que pueden variar dependiendo de factores como la distribución, o los parámetros utilizados al compilar el kernel. En consecuencia, sí pueden ser modificados por el usuario a su conveniencia, dependiendo de sus necesidades. Puedes encontrar más información sobre una forma de realizar lo anterior en [este enlace](https://unix.stackexchange.com/questions/466722/how-to-change-the-length-of-time-slices-used-by-the-linux-cpu-scheduler). ¡Espero haber respondido a tu pregunta!
